common/eth2/src/lib.rs

```diff
@@ -395,6 +408,34 @@ impl BeaconNodeHttpClient {
         ok_or_error(response).await
     }
 
+    /// Generic POST function supporting arbitrary responses and timeouts.
+    /// Does not include Content-Type application/json in the request header.
+    /// Does include Eth-Consensus-Version in the request header.
+    async fn post_generic_json_without_content_type_header_but_with_consensus_header<
+        T: Serialize,
+        U: IntoUrl,
+    >(
+        &self,
+        url: U,
+        body: &T,
+        timeout: Option<Duration>,
+        fork: ForkName,
+    ) -> Result<Response, Error> {
+        let mut builder = self.client.post(url);
+        if let Some(timeout) = timeout {
+            builder = builder.timeout(timeout);
+        }
+
+        let serialized_body = serde_json::to_vec(body).map_err(Error::InvalidJson)?;
```

Comment:
I didn't get why we are doing it this way for attester slashings. Why not just post the json with the content header?

Reply:
i sort of just went with what was there already. I actually think this was just included for testing this PR and should have been reverted https://github.com/sigp/lighthouse/pull/4575

Reply:
updated 71a2eadc467040a4a2996177e557673e5dc104da